ludwig/utils/schema_utils.py

```diff
@@ -159,6 +346,75 @@ def _jsonschema_type_mapping(self):
         }
 
 
+def IntegerOrStringOptionsField(
+    options: List[str],
+    nullable: bool,
+    default: Union[None, int],
+    is_integer: bool = True,
+    min: Union[None, int] = None,
+    max: Union[None, int] = None,
+    min_exclusive: Union[None, int] = None,
+    max_exclusive: Union[None, int] = None,
+):
+    is_integer = True
+    return NumericOrStringOptionsField(**locals())
+
+
+def NumericOrStringOptionsField(
```

Comment:
The definitions for these custom Marshmallow fields are actually quite long!

It might be useful to add small unit tests for our custom primitive-level Marshmallow validators (like `NumericOrStringOptionsField`) to concisely demonstrate what they do and don't allow, like:

```
@dataclass
class Config:
    parameter: float = schema.NumericOrStringOptionsField(
        default=0.001, min=0.0, max=1.0, options=["auto"], nullable=False
    )

Config(parameter=0.1)
with pytest.raises(ValidationError):
    Config(parameter="not")
with pytest.raises(ValidationError):
    Config(parameter="2.0")
assert Config().parameter == 0.0001
```

In addition, could we add Docstrings explaining each of the different fields and what validation behavior they control/enforce/govern?

It might be a bit tedious now, but I think we'll be happy in a few months that the different parameters of our custom validators are all documented/tested. WDYT?

Reply:
Yup, working on unit tests right now

Reply:
And yes, agreed on explaining the validators

Reply:
added